Ports sound synthesizer from VG (with a few changes)

ModDownloading opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

Ports the sound synthesizer item from VG (and Bee) as a new toy in the theater's toy chest. 

It can produce a lot of different sounds including applause and laugh tracks.

WARNING: THIS IS VERY MUCH IN PROGRESS AND STILL NEEDS APPROVAL FOR NEW SOUNDS, DO NOT MERGE THIS YET.
I'M JUST MAKING THIS PUBLIC NOW SO RADBURY KNOWS I'M WORKING ON IT.

New sounds are: a laughtrack, applause, a rimshot sound (like a cartoon ricochet) and slide whistle noises. I cut a few sounds like the one that screams YEET or the AI's "just like one of those Japanese animes" quote, but I can add them back in if anyone wants them and can get approval from the server staff.

## Why It's Good For The Game

@RaveRadbury wanted a way to produce a theater-audience style applause and laughing after my other PR added audio cooldowns to both of those, so here you go!

## Changelog
:cl:
add: The amazing sound synthesizer! Capable of over 10 different sounds! Get it at your local theater in the toy chest (or off of the clown after you mug him!).
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

ModDownloading committed:
Adds sound_synth file (but fixes variable names)

ModDownloading committed:
Adds some of the missing soundfiles

ModDownloading committed:
actually adds the sound synthesizer to the toy chest

interestingusernam3 commented:
FYI this does not compile

ModDownloading commented:
@interestingusernam3 It's still in progress, I marked it as a draft for a reason. It is definitely not ready for review yet.

RaveRadbury commented:
Making good on your word. Thank you.

ModDownloading committed:
Rename variables into TG equivalents

ModDownloading committed:
Changes the icon to something we actually have

ModDownloading committed:
Almost forgot to update tgstation.dme

ModDownloading committed:
Delete sound_effects_laughtrack.ogg

ModDownloading committed:
Delete sound_effects_applause.ogg

ModDownloading committed:
Delete sound_effects_rimshot.ogg

ModDownloading committed:
Delete sound_effects_slide_whistle_long.ogg

ModDownloading committed:
Delete sound_effects_slide_whistle_short.ogg

ModDownloading committed:
re-add the files WITH THE CORRECT NAMES THIS TIME

ModDownloading committed:
Hopefully fixes indentations

ModDownloading commented:
Ok I tested it and it's working as intended, ready for review!

Fikou commented on code/game/objects/items/devices/sound_synth.dm:
```diff
@@ -0,0 +1,64 @@
+/obj/item/device/soundsynth
```
why is this a device? does that subtype even exist?

ModDownloading replied:
Nope, can't believe I didn't realize stuff like multitools weren't devices despite being under the device folder, it's fixed now!

Fikou commented on code/game/objects/items/devices/sound_synth.dm:
```diff
@@ -0,0 +1,64 @@
+/obj/item/device/soundsynth/attack_self(mob/user as mob)
+	if(spam_flag + 2 SECONDS < world.timeofday)
+		playsound(src, selected_sound, volume, shiftpitch)
+		spam_flag = world.timeofday
```
why are you using timeofday instead of time?

ModDownloading replied:
Fixed!

Fikou commented on code/game/objects/items/devices/sound_synth.dm:
```diff
@@ -0,0 +1,64 @@
+		spam_flag = world.timeofday
+
+/obj/item/device/soundsynth/AltClick()
+	if(!usr.incapacitated() && Adjacent(usr))
```
dont use usr, altclick has an user arg

ModDownloading replied:
Deleted both usr sections, while keeping the functions, will have to test again to make sure it continues to work properly.

Fikou commented on code/game/objects/items/devices/sound_synth.dm:
```diff
@@ -0,0 +1,64 @@
+	icon = 'icons/obj/device.dmi'
+	w_class = WEIGHT_CLASS_TINY
+
+	var/tmp/spam_flag = 0 //To prevent mashing the button to cause annoyance like a huge idiot.
```
tmp?

Fikou replied:
just use the cooldown defines for this not weird stuff

ModDownloading replied:
Which cooldown should I use? I think I found the defines but none of them seem to be 2 seconds long.

If I'm supposed to use the defines to create a 2 second long cooldown (or longer), what exactly is the proper syntax to do that?

ModDownloading commented:
Alright, making adjustments, I didn't know which parts of vg code were fine and which parts needed improvement, so thank you for pointing those out. Please don't send the clown after me, I was just kidding about the mugging thing.

ModDownloading committed:
Update sound_synth.dm

Removed device subtype, removed "as mob" and usr from attack and AltClick respectively, changed spam_flag to playback_cooldown and changed timeofday to time.

ModDownloading committed:
Removed device subtype

ModDownloading commented on code/game/objects/items/devices/sound_synth.dm:
```diff
@@ -0,0 +1,64 @@
+	"Time Stop" = "selected_sound=sound/magic/timeparadox2.ogg&shiftpitch=0&volume=80",
+	)
+
+/obj/item/device/soundsynth/verb/pick_sound()
```
What would you recommend I make it instead of a verb? AltClick and attackself are taken already.

ModDownloading commented on code/game/objects/items/devices/sound_synth.dm:
```diff
@@ -0,0 +1,64 @@
+		playsound(src, selected_sound, volume, shiftpitch)
+		spam_flag = world.timeofday
+
+/obj/item/device/soundsynth/attack(mob/living/M as mob, mob/living/user as mob, def_zone)
```
Changed the spam_flag name to playback_cooldown and removed the "as mob"s.

Mothblocks commented:
> You currently have a negative Fix/Feature pull request delta of -8. Maintainers may close this PR at will. Fixing issues or improving the codebase will improve this score.

ModDownloading commented:
@Mothblocks I was working on this PR as a favor for @RaveRadbury, and I was going to head to the coder channel to get help with bringing the code up to TG standards. If the issue is with the code, but not the content, could you please reopen this while I go to fix it? If the issue is with the content (the sound synthesizer itself), are there any changes you would like me to make to it to make it satisfactory?

Mothblocks commented:
Fixing issues or improving the codebase will improve this score. 😛 

File bug fix PRs, quality of life, etc.
